### ISSUE TITLE
Saving my finetuned model 

### ISSUE THREAD

**Issue by solablue** [Lightning-AI/lit-gpt#340]
Hello, I was able to fine-tune Llama2 with the help of lit-gpt. Thanks! However, there seems to be **a mismatch between the keys in the state dictionary I am trying to load and the keys in the model's state dictionary**. My ultimate goal is to upload my model to Huggingface space. Upon running the code, I encountered the following error:

```
import transformers
from transformers import LlamaForCausalLM, LlamaTokenizer

# Load Tokenizer
tokenizer = LlamaTokenizer.from_pretrained('/content/lit-parrot/checkpoints/meta-llama/Llama-2-7b-chat-hf')

# Load Model
model = LlamaForCausalLM.from_pretrained('meta-llama/Llama-2-7b-chat-hf')
checkpoint = torch.load('/version_5/lit_model_adapter_finetuned.pth', map_location=torch.device('cpu'))
model_state_dict = checkpoint['model']
model.load_state_dict(model_state_dict)
torch.save(checkpoint['model'], '/checkpoints/model_weights.pth')

# model = LlamaForCausalLM.from_pretrained('/version_5/lit_model_adapter_finetuned.pth')
```

```
[/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py](https://localhost:8080/#) in load_state_dict(self, state_dict, strict, assign)
   2150 
   2151         if len(error_msgs) > 0:
-> 2152             raise RuntimeError('Error(s) in loading state_dict for {}:\n\t{}'.format(
   2153                                self.__class__.__name__, "\n\t".join(error_msgs)))
   2154         return _IncompatibleKeys(missing_keys, unexpected_keys)

RuntimeError: Error(s) in loading state_dict for LlamaForCausalLM:
	Missing key(s) in state_dict: "model.embed_tokens.weight", "model.layers.0.self_attn.q_proj.weight", "model.layers.0.self_attn.k_proj.weight", "model.layers.0.self_attn.v_proj.weight", "model.layers.0.self_attn.o_proj.weight", "model.layers.0.self_attn.rotary_emb.inv_freq", "model.layers.0.mlp.gate_proj.weight", "model.layers.0.mlp.up_proj.weight", "model.layers.0.mlp.down_proj.weight", "model.layers.0.input_layernorm.weight", "model.layers.0.post_attention_layernorm.weight", "model.layers.1.self_attn.q_proj.weight", "model.layers.1.self_attn.k_proj.weight", "model.layers.1.self_attn.v_proj.weight", "model.layers.1.self_attn.o_proj.weight", "model.layers.1.self_attn.rotary_emb.inv_freq", "model.layers.1.mlp.gate_proj.weight", "model.layers.1.mlp.up_proj.weight", "model.layers.1.mlp.down_proj.weight", "model.layers.1.input_layernorm.weight", "model.layers.1.post_attention_layernorm.weight", "model.layers.2.self_attn.q_proj.weight", "model.layers.2.self_attn.k_proj.weight", "model.layers.2.self_attn.v_proj.weight", "model.layers.2.self_attn.o_proj.weight", "model.layers.2.self_attn.rotary_emb.inv_freq", "model.layers.2.mlp.gate_proj.weight", "model.layers.2.mlp.up_proj.weight", "model.layers.2.mlp.down_proj.weight", "model.layers.2.input_layernorm.weight", "model.layers.2.post_attention_layernorm.weight", "model.layers.3.self_attn.q_proj.weight", "model.layers.3.self_attn.k_proj.weight", "model.layers.3.self_attn.v_proj.weight", "model.layers.3.self_attn.o_proj.weight", "mo...
	Unexpected key(s) in state_dict: "transformer.wte.weight", "transformer.h.0.norm_1.weight", "transformer.h.0.attn.attn.weight", "transformer.h.0.attn.proj.weight", "transformer.h.0.norm_2.weight", "transformer.h.0.mlp.fc_1.weight", "transformer.h.0.mlp.fc_2.weight", "transformer.h.0.mlp.proj.weight", "transformer.h.1.norm_1.weight", "transformer.h.1.attn.attn.weight", "transformer.h.1.attn.proj.weight", "transformer.h.1.norm_2.weight", "transformer.h.1.mlp.fc_1.weight", "transformer.h.1.mlp.fc_2.weight", "transformer.h.1.mlp.proj.weight", "transformer.h.2.norm_1.weight", "transformer.h.2.attn.gating_factor", "transformer.h.2.attn.attn.weight", "transformer.h.2.attn.proj.weight", "transformer.h.2.attn.adapter_wte.weight", "transformer.h.2.norm_2.weight", "transformer.h.2.mlp.fc_1.weight", "transformer.h.2.mlp.fc_2.weight", "transformer.h.2.mlp.proj.weight", "transformer.h.3.norm_1.weight", "transformer.h.3.attn.gating_factor", "transformer.h.3.attn.attn.weight", "transformer.h.3.attn.proj.weight", "transformer.h.3.attn.adapter_wte.weight", "transformer.h.3.norm_2.weight", "transformer.h.3.mlp.fc_1.weight", "transformer.h.3.mlp.fc_2.weight", "transformer.h.3.mlp.proj.weight", "transformer.h.4.norm_1.weight", "transformer.h.4.attn.gating_factor", "transformer.h.4.attn.attn.weight", "transformer.h.4.attn.proj.weight", "transformer.h.4.attn.adapter_wte.weight", "transformer.h.4.norm_2.weight", "transformer.h.4.mlp.fc_1.weight", "transformer.h.4.mlp.fc_2.weight", "transformer.h...
```

**Comment by JustinGoheen** [comment]
Hi @solablue!  After finetuning, we need to convert the Lit-GPT model format back to the original model format. For instance, if you did full finetuning, you would run the following:

```sh
python scripts/convert_lit_checkpoint.py \
    --checkpoint_name lit_model_finetuned.pth \
    --checkpoint_dir checkpoints/meta-llama/Llama-2-7b-chat-hf
```

Please see the tutorial for [converting Lit-GPT models](https://github.com/Lightning-AI/lit-gpt/blob/main/tutorials/convert_lit_models.md) for more information, and please let me know if this helps.

**Comment by solablue** [comment]
Thanks. I follows your instruction but unfortunately got this error. Don't you support 7b chat model? 

`NotImplementedError: Conversion for Llama-2-7b-chat-hf is not yet supported
`

**Comment by stephen-itschner** [comment]
In the same boat as @solablue. Getting the trained llama model backed out again from this custom format will be critical to my project

**Comment by JustinGoheen** [comment]
@solablue @stephen-itschner this will be resolved in #315 and #341.
